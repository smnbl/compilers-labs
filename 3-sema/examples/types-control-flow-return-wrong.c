int foo()
{
    return 2.0;
}

// Another wrong example:
// void foo()
// {
//     return 2.0;
// }
