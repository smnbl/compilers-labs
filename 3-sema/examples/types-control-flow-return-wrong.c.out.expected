sema: error: Type of return statement does not match function return type
└── Program <4>
    └── FuncDecl: returnType = 'int', name = 'foo' <3>
        └── CompoundStmt <2>
            └── ReturnStmt <1>
                └── FloatLiteral: value = '2' <0>
